### PR TITLE
feat: theme unification and cleaning

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

## Theme unification

We used to have one `styled-components` theme which contained styles for all apps mixed.
We should separate common and app-specific styles.

Common components and theme are located in `libs/ui`.
The common theme is presented as `baseTheme()` and its interface `CowProtocolTheme`.
Values defined in this theme are common, it means they can be used in any library or application.

Each app now has `src/theme` directory which should contain two mandatory entities:
 - `ThemeProvider`
 - `declare module 'styled-components'`

Each app can extend common theme (`CowProtocolTheme`) via `declare module`:

```
declare module 'styled-components' {
  export interface MyAppTheme extends CowProtocolTheme {
    myAppSpecificVariable: boolean
  }

  export interface DefaultTheme extends MyAppTheme {}
}
```

This is the main magic. With this declaration you  can use all variables from `CowProtocolTheme` + variables from `MyAppTheme`.

## Theme cleaning

I've removed many values from `CowProtocolTheme` or moved them to a specific component in case when they are in use only there.
Now `CowProtocolTheme` contains two sets of values: `Colors` and `ThemeUtils`.

# To Test

1. CowSwap should look as before
2. Explorer should look as before
3. Cow.fi should look as before

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -262,7 +262,6 @@
     "@types/ms": "^0.7.31",
     "@types/ms.macro": "^2.0.0",
     "@types/node": "18.16.9",
-    "@types/react": "18.2.14",
```
Comment (shoom3301) [inline package.json:265]: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27805#issuecomment-543838377

**Comment by shoom3301** [comment]
Now I feel like:

image

**Comment by elena-zh** [comment]
Hi @shoom3301 , great job!
Some places on Explorer with incorrect icons: 
1. Menu 
image
2. Loading cow 
image
3. icons in the batch graph 
image


Other than that, changes LGTM!

**Comment by shoom3301** [comment]
Hi @elena-zh, those issues are not from this PR, they are already in https://dev.explorer.cow.fi

**Review by elena-zh — APPROVED** [review]
Moved  these issues to the [doc ](https://www.notion.so/cownation/Rebrand-tests-CoW-Swap-deb5ca44d8bd4ce8bed18c674f5a8135?pvs=4#0f9b9a0f92c542829e64bd2371a35042)

File: apps/cow-fi/theme/ThemeProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { PropsWithChildren, useMemo } from 'react'
+
+import './types'
+
+function getCowfiTheme() {
+  return baseTheme(THEME_MODE)
+}
+
+export function ThemeProvider(props: PropsWithChildren) {
+  const theme = useMemo(() => getCowfiTheme(), [])
```
Comment (alfetopito) [inline apps/cow-fi/theme/ThemeProvider.tsx:13]: Don't we need to disable the es-lint rule for this to pass the linter?

File: apps/cowswap-frontend/src/legacy/components/swap/UnsupportedCurrencyFooter/UnsupportedCurrencyFooterMod.tsx [hunk 3]
```diff
@@ -8,10 +8,11 @@ import { Currency } from '@uniswap/sdk-core'
 
 import { Trans } from '@lingui/macro'
 import styled from 'styled-components/macro'
+import { ThemedText, Z_INDEX } from 'theme'
+import { CloseIcon } from 'theme'
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/swap/UnsupportedCurrencyFooter/UnsupportedCurrencyFooterMod.tsx:12]: Nitpick: unify imports in a single line

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/styled.ts [hunk 4]
```diff
@@ -383,10 +382,10 @@ export const TransactionInnerDetail = styled.div`
 
 export const TextAlert = styled.div<{ isPending: boolean; isExpired: boolean; isCancelled: boolean }>`
   background: ${({ theme, isPending }) =>
-    isPending ? transparentize(theme.attention, 0.85) : transparentize(theme.success, 0.85)};
+    isPending ? transparentize('#ff5722', 0.85) : transparentize(theme.success, 0.85)};
   margin: 6px 0 16px;
   padding: 8px 12px;
-  color: ${({ theme, isPending }) => (isPending ? theme.attention : theme.success)};
+  color: ${({ theme, isPending }) => (isPending ? '#ff5722' : theme.success)};
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/styled.ts:388]: Why the hardcoded values?

**Review by alfetopito — APPROVED** [review]
Looks good on desktop firefox.
Some minor comments

File: apps/cow-fi/theme/ThemeProvider.tsx (hunk 2 above)
Comment (shoom3301) [reply apps/cow-fi/theme/ThemeProvider.tsx:13]: No, there are no dependencies. `getCowfiTheme` is a pure function

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/styled.ts (hunk 4 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/account/containers/Transaction/styled.ts:388]: The color is only used in this file, it shouldn't be in theme config.
I assume it must be replaced by some alternative from theme, but decided not to do it right now.
In the future we should make a search `/\#[\da-zA-Z]{3,6}/` and review all hardcoded values with Michel.

File: apps/cow-fi/styles/global.styles.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import styled, { createGlobalStyle } from 'styled-components'
```
Comment (anxolin) [inline apps/cow-fi/styles/global.styles.ts:1]: We had a lint rule before to prevent using 'styled-components', maybe doesn't affect cow.fi?

**Pull request merged**